### ISSUE TITLE
[deeplab] training dataset without pretrained models

### ISSUE THREAD

**Issue by flavea** [tensorflow/models#4204]
### System information
- **What is the top-level directory of the model you are using**: deeplab
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: pip install
- **TensorFlow version (use command below)**: 1.6.0
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: -
- **GPU model and memory**: NVIDIA GeForce GTX 1060 6GB
- **Exact command to reproduce**:

### Describe the problem
Hello, I am currently trying to train an aerial/satellite dataset ([Volodymyr Mnih's dataset](https://www.cs.toronto.edu/~vmnih/data)) for building/road detection using deeplab as a research. I am wondering if it's possible to train the data without pre-trained models, because it seemed that the pre-trained models available are not suited for aerial/satellite images. I tried to train the data without using any pre-trained model, setting tf_initial_checkpoint as not required, but when I test the training result, nothing is really detected .  After reading around the issues and the documentation, I have set initialize_last_layer and last_layers_contain_logits_only to false to ignore the classes from the pre-trained model. I got better results, but the result is very inaccurate.

I am still checking if it's a problem with the dataset or if it's because the pre-trained model is incompatible with this dataset. But I am still wondering if I set the flags right and is it possible to use this code without any pre-trained model?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

flags: python train.py \ --logtostderr \ --training_number_of_steps=1000 \ --train_split="train" \--model_variant="xception_65" \ --atrous_rates=6 \ --atrous_rates=12 \ --atrous_rates=18 \ --output_stride=16 \ --decoder_output_stride=4 \ --train_crop_size=513 \ --train_crop_size=513 \ --train_batch_size=2 \ --resize_factor=16 \ --fine_tune_batch_norm=False \ --dataset="mnih" \ --initialize_last_layer=False \ --last_layers_contain_logits_only=True \ --train_logdir="datasets/mnih_buildings/exp/train_on_train_set/train/" \ --dataset_dir="datasets/mnih_buildings/tfrecord/" \ --tf_initial_checkpoint="datasets/pascal_voc_seg/init_models/deeplabv3_pascal_train_aug/model.ckpt" 
<img src="https://i.imgur.com/aRAAgxs.png">

flags: python train.py \ --logtostderr \ --training_number_of_steps=1000 \ --train_split="train" \--model_variant="xception_65" \ --atrous_rates=6 \ --atrous_rates=12 \ --atrous_rates=18 \ --output_stride=16 \ --decoder_output_stride=4 \ --train_crop_size=513 \ --train_crop_size=513 \ --train_batch_size=2 \ --resize_factor=16 \ --fine_tune_batch_norm=False \ --dataset="mnih" \ --initialize_last_layer=False \ --last_layers_contain_logits_only=False \ --train_logdir="datasets/mnih_buildings/exp/train_on_train_set/train/" \ --dataset_dir="datasets/mnih_buildings/tfrecord/" \ --tf_initial_checkpoint="datasets/pascal_voc_seg/init_models/deeplabv3_pascal_train_aug/model.ckpt" 
<img src="https://i.imgur.com/QIo0IKA.png">

flags: python train.py \ --logtostderr \ --training_number_of_steps=1000 \ --train_split="train" \--model_variant="xception_65" \ --atrous_rates=6 \ --atrous_rates=12 \ --atrous_rates=18 \ --output_stride=16 \ --decoder_output_stride=4 \ --train_crop_size=513 \ --train_crop_size=513 \ --train_batch_size=2 \ --resize_factor=16 \ --fine_tune_batch_norm=False \ --dataset="mnih" \ --initialize_last_layer=False \ --last_layers_contain_logits_only=False \ --train_logdir="datasets/mnih_buildings/exp/train_on_train_set/train/" \ --dataset_dir="datasets/mnih_buildings/tfrecord/"
<img src="https://i.imgur.com/JrxB3NT.png">

**Comment by aquariusjay** [comment]
Can you try using the provided checkpoint as the initial_checkpoint?
Even thought the pretrained dataset has different statistics from yours, it is still better to use that (better than randomly initialization).

**Comment by tonghe90** [comment]
@aquariusjay this is the parameters
python deeplab/train.py \
    --logtostderr \
    --base_learning_rate=0.007 \
    --num_clones=4 \
    --training_number_of_steps=30000 \
    --learning_power=0.9 \
    --model_variant="xception_65" \
    --atrous_rates=6 \
    --atrous_rates=12 \
    --atrous_rates=18 \
    --output_stride=16 \
    --decoder_output_stride=4 \
    --train_crop_size=513 \
    --train_crop_size=513 \
    --train_batch_size=16 \
    --fine_tune_batch_norm=True \
    --dataset="pascal_voc_seg" \
    --last_layer_gradient_multiplier=10 \
    --train_split="trainaug" \
    --tf_initial_checkpoint="./deeplab/xception/model.ckpt" \
    --train_logdir="./deeplab/trainlogs2/trainaug" \
    --dataset_dir="./deeplab/datasets/pascal_voc_seg/tfrecord"

**Comment by aquariusjay** [comment]
@tonghe90 
Could you please open a new issue as your question is not related to the original one?
Regarding your question, you are only computing the batch norm statistics with (16 / 4) 4 batch size for each clone. We use 16 batch size for each clone in our experiments.

**Comment by arifiany** [comment]
@aquariusjay I have used the pascal checkpoint in the above example result (the second one). There are results, but at around 280-step, I ran to Tensor had NaN values error

**Comment by mahxn0** [comment]
@flavea  did you solve this question?    i meet the same error, predict result is all 00000000000000000000000000000000000000000000000000000000000000000000000000 ,  can  help me ?  thank you
![figure_1-1](https://user-images.githubusercontent.com/19154155/49929560-c16b0f00-fefd-11e8-9864-569670e69fc7.png)

**Comment by sidgan** [comment]
@flavea @mahxn0 were you able to resolve this?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
